src/cats/SlideAnimation.js

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from "react";
+import "./SlideAnimation.css";
+
+const Direction = {
+  Next: 0,
+  Previous: 1,
+};
+
+function SlideAnimation({ child, direction }) {
+  const [slideChildren, setSlideChildren] = useState([]);
+
+  useEffect(() => {
+    setSlideChildren((prevChildren) => {
```

Comment:
this is a bit hard to follow, but I've wrote a similar component recently, `Crossfade` and this logic is somewhat complicated anyway